proto/proto/note.proto

```diff
@@ -28,8 +28,10 @@ message NoteMetadata {
     fixed64 aux = 5;
 }
 
-// Represents a note.
-message Note {
+// Represents a committed note.
+//
+// A committed note is a note that has been included in a block.
+message CommittedNote {
```

Comment:
I would replace `block_num`, `note_index`, `note_id`, and `merkle_path` with `NoteInclusionInBlockProof`.

We could also try to replace `metadata` and `details` with the `Note`, but this may be more tricky as `details` are optional here and mandatory there.

Reply:
Something to maybe consider is how this differs from a `NoteSyncRecord`?

I imagine we technically only have notes and committed notes. And if we want less types then we can collapse them all into a singular type with optional fields.

Reply:
I agree, it might we might be losing more type safety than needed. In the other hand, it really helps to reduce the proto messages.